Subject: only compile with -O1 if the compiler is broken
From: Olaf Hering <olh@suse.de>
Patch-mainline: 2.6.19

Only compile with -O1 if the (very old) compiler is broken.  We use
reiserfs alot since SLES9 on ppc64, and it was never seen with gcc33. 
Assume the broken gcc is gcc-3.4 or older.

Signed-off-by: Olaf Hering <olh@suse.de>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 fs/reiserfs/Makefile |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff -puN fs/reiserfs/Makefile~use-gcc-o1-in-fs-reiserfs-only-for-ancient-gcc-versions fs/reiserfs/Makefile
--- a/fs/reiserfs/Makefile~use-gcc-o1-in-fs-reiserfs-only-for-ancient-gcc-versions
+++ a/fs/reiserfs/Makefile
@@ -28,7 +28,7 @@ endif
 # will work around it. If any other architecture displays this behavior,
 # add it here.
 ifeq ($(CONFIG_PPC32),y)
-EXTRA_CFLAGS := -O1
+EXTRA_CFLAGS := $(call cc-ifversion, -lt, 0400, -O1)
 endif
 
 TAGS:
_
